sno/commit.py

```diff
@@ -147,5 +193,29 @@ def get_commit_message(repo, wc_changes):
     # - whitespace at start/end
     # - comment lines
     # - blank lines surrounding comment lines
-    message = re.sub(r"^\n*#.*\n", "", message.strip(), flags=re.MULTILINE)
+    message = re.sub(r"^\n*#.*\n", "", message, flags=re.MULTILINE)
     return message.strip()
+
+
+def commit_output_to_text(jdict):
+    jdict = jdict["sno.commit/v1"]
+    branch = jdict["branch"]
+    commit = jdict["abbrevCommit"]
+    message = jdict["message"].replace("\n", " ")
+    diff = diff_status_to_text(jdict["changes"])
+    datetime = to_long_local_iso8601(from_short_iso8601(jdict["commitTime"]))
+    return f"[{branch} {commit}] {message}\n{diff}\n  Date: {datetime}"
+
+
+def to_short_iso8601(datetime):
+    """Returns a string like: 2020-03-26T09:10:11Z"""
+    return datetime.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+def from_short_iso8601(s):
+    return datetime.fromisoformat(s.replace("Z", "+00:00"))
+
+
+def to_long_local_iso8601(datetime):
+    """Returns a string like: 2020-03-26 21:10:11 +12:00"""
```

Comment:
It's valid ISO8601 with or without the `T`/`Z`/etc, so why not have human (ctime) and iso8601, rather than two flavours of the latter? For human timestamps, If we wanted to take into account `Commit.commit_time_offset` too:

```python
def human_committer_time(commit_time, commit_time_offset):
    class CommitTZ(tzinfo):
        def utcoffset(self, dt): return timedelta(minutes=commit_time_offset)
    return datetime.utcfromtimestamp(timestamp).astimezone(CommitTZ()).strftime("%c %z")
```

(I _think_ git normally displays committer-times rather than local times?)

Reply:
Now including commit_time_offset in json output, and use it for formatting the time in text output.